EventFilter/GEMRawToDigi/interface/GEBdata.h

```diff
@@ -0,0 +1,138 @@
+      m_ErrorC = 0b0001111111111111 & (word);    /*!<Thirteen Flags*/
+      for(int i=0; i<13; ++i)
+	{
+	  m_GEBflags.push_back(0x01 & (m_ErrorC >> i));
+	}
+    }
+    uint64_t getChamberHeader() const
+    {
+      return
+	(static_cast<uint64_t>(m_ZeroSup & 0x00ffffff) <<  40) |
+	(static_cast<uint64_t>(m_InputID & 0b00011111) <<  35) |
+	(static_cast<uint64_t>(m_Vwh & 0x0fff) <<  23) |
+	(static_cast<uint64_t>(m_ErrorC & 0b0001111111111111));
+    }
+
+    //return specific flags
+    //!Returns one of thirteen flags from GEM chamber header.
+    /**
+       Argument must be between 0 and 12. The flags corresponding to a given argument are shown.
+       12->EvtFIFO full    11->InFIFO full    10->L1AFIFO full   9->Even size overflow    8->EvtFIFO near full   5->InFIFO near full    
+       6->L1AFIFO near full    5->Event size warn   4->No VFAT marker    3->OOS GLIB VFAT   2->OOS GLIB OH 
+       1->BX mismatch GLIB VFAT    0->BX mismatch GLIB OH
+    */
+    uint8_t getGEBflag(int c) const
+    {
+      return m_GEBflags.at(c);
+    }
+    std::vector<uint8_t> getGEBflag() const
+    {
+      return m_GEBflags;
+    }    
+    // need to include all the flags
+    //!Reads the word for GEM Chamber Trailer
+    /**
+       Fills the OH CRC, VFAT word count, InFIFO underflow, and Stuck data.
+    */
+    void setChamberTrailer(uint64_t word)
+    {
+      m_OHCRC = word >> 48;           /*!<OH CRC*/
+      m_Vwt = 0x0fff & (word >> 36);  /*!<VFAT word count*/
+      m_InFu = 0x0f & (word >> 35);   /*!<InFIFO underflow*/
+      m_Stuckd = 0x01 & (word >> 34); /*!<Stuck data*/
+    }
+    uint64_t getChamberTrailer() const
+    {
+      return
+	(static_cast<uint64_t>(m_OHCRC) <<  48) |
+	(static_cast<uint64_t>(m_Vwt & 0x0fff) <<  36) |
+	(static_cast<uint64_t>(m_InFu & 0x0f) <<  35) |
+	(static_cast<uint64_t>(m_Stuckd & 0x01) << 34);
+    }
+
+    void setVwh(uint16_t n){m_Vwh = n;}       ///<Returns VFAT word count (size of VFAT payload)
+    void setInputID(uint8_t n){m_InputID = n;}///<Returns GLIB input ID
+
+    uint32_t zeroSup()  const {return m_ZeroSup;}   ///<Returns Zero Suppression flags
+    uint8_t  inputID()  const {return m_InputID;}   ///<Returns GLIB input ID
+    uint16_t vwh()      const {return m_Vwh;}       ///<Returns VFAT word count (size of VFAT payload)
+    uint16_t errorC()   const {return m_ErrorC;}    ///<Returns thirteen flags in GEM Chamber Header
+    
+    uint16_t ohCRC()    const {return m_OHCRC;}     ///<Returns OH CRC 
+    uint16_t vwt()      const {return m_Vwt;}       ///<Returns VFAT word count
+    uint8_t  inFu()     const {return m_InFu;}      ///<Returns InFIFO underflow flag
+    uint8_t  stuckd()   const {return m_Stuckd;}    ///<Returns Stuck data flag
+
+    //!Adds VFAT data to the vector
+    void addVFAT(VFATdata v){m_vfatd.push_back(v);}
+    //!Returns the vector of FVAT data
+    std::vector<VFATdata> vFATs() const {return m_vfatd;}  
```

Comment:
return a ref?
same in some other places

Reply:
should i go ref or pointers?

Reply:
although equivalent, vectors of structs are normally returned by ref (and yiu don't hve to touch the rest of your code if you do so)